pennylane_lightning/lightning_qubit/_state_vector.py

```diff
@@ -0,0 +1,386 @@
+                consisting of 0s and 1s.
+            wires (Wires): wires that the provided computational state should be
+                initialized on
+
+        Note: This function does not support broadcasted inputs yet.
+        """
+        num = self._get_basis_state_index(state, wires)
+        self._create_basis_state(num)
+
+    def _apply_lightning_controlled(self, operation):
+        """Apply an arbitrary controlled operation to the state tensor.
+
+        Args:
+            operation (~pennylane.operation.Operation): operation to apply
+
+        Returns:
+            array[complex]: the output state tensor
+        """
+        state = self.state_vector
+
+        basename = "PauliX" if operation.name == "MultiControlledX" else operation.base.name
+        if basename == "Identity":
+            return
+        method = getattr(state, f"{basename}", None)
+        control_wires = self.wires.indices(operation.control_wires)
+        control_values = (
+            [bool(int(i)) for i in operation.hyperparameters["control_values"]]
+            if operation.name == "MultiControlledX"
+            else operation.control_values
+        )
+        if operation.name == "MultiControlledX":
+            target_wires = list(set(self.wires.indices(operation.wires)) - set(control_wires))
+        else:
+            target_wires = self.wires.indices(operation.target_wires)
+        if method is not None:  # apply n-controlled specialized gate
+            inv = False
+            param = operation.parameters
+            method(control_wires, control_values, target_wires, inv, param)
+        else:  # apply gate as an n-controlled matrix
+            method = getattr(state, "applyControlledMatrix")
+            target_wires = self.wires.indices(operation.target_wires)
+            try:
+                method(
+                    qml.matrix(operation.base),
+                    control_wires,
+                    control_values,
+                    target_wires,
+                    False,
+                )
+            except AttributeError:  # pragma: no cover
+                # To support older versions of PL
+                method(operation.base.matrix, control_wires, control_values, target_wires, False)
+
+    def apply_lightning(self, operations):
+        """Apply a list of operations to the state tensor.
+
+        Args:
+            operations (list[~pennylane.operation.Operation]): operations to apply
+
+        Returns:
+            array[complex]: the output state tensor
+        """
+        state = self.state_vector
+
+        # Skip over identity operations instead of performing
+        # matrix multiplication with it.
+        for operation in operations:
+            name = operation.name
+            if name == "Identity":
+                continue
+            method = getattr(state, name, None)
+            wires = self.wires.indices(operation.wires)
```

Comment:
Given the wires are now just [0,1,2,3,4, ...], I think we can just do `wires = list(operation.wires)`.